src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableEntityContext.cs

```diff
@@ -144,14 +158,19 @@ public bool ErrorsPresent(out string description)
             }
         }
 
+        public void Rollback(int position)
+        {
+            this.CurrentStateAccess = StateAccess.NotAccessed;
+            this.CurrentState = null;
+            this.outbox.RemoveRange(position, this.outbox.Count - position);
```

Comment:
It might be worth adding comments here explaining why this works as a means of rolling back the operation(s). My assumption based on our conversations is that setting `CurrentState` to `null` basically causes us to not write back the updates to storage. Is that accurate? And I assume removing items from the outbox effectively discards side effects such as sending signals to other entities?

Reply:
Yes, your assumptions are correct. I can add some comments there. Essentially, `this.CurrentState` (which can be an object) is the current state of the entity in memory, and gets modified while an operation executes, while `this.State.EntityState` (which is a string) represents the state of the entity before the operation begins.